src/wp-includes/class-wp-scripts.php

```diff
@@ -1148,6 +989,11 @@ private function get_eligible_loading_strategy( $handle ) {
 			return '';
 		}
 
+		// Handles with inline scripts attached in the 'after' position cannot be delayed.
+		if ( $this->has_inline_script( $handle, 'after' ) ) {
+			return '';
+		}
+
 		// Handle async strategy scenario.
 		if ( 'async' === $intended_strategy && $this->has_only_delayed_dependents( $handle, true ) ) {
```

Comment:
I believe we also have to update the `has_only_delayed_dependents()` method to return `false` if _any_ dependent has an inline "after" script.

Basically for every dependent handle you also have to call the above `has_inline_script( $handle, 'after' )`.